plugins/modules/net_tools/nmcli.py

```diff
@@ -1203,19 +1219,8 @@ def connection_options(self, detect_change=False):
             })
         elif self.type == 'wifi':
             options.update({
-                '802-11-wireless.ssid': self.ssid,
                 'connection.slave-type': 'bond' if self.master else None,
```

Comment:
As per the issue thread I'd prefer a new option be introduced to set the "client"-type instead of trying to leverage the existing handling of the `type` option. If that creates redundancy, setting the `bridge-port` options for example, then it would make sense to pull that functionality out and make it reusable.

Reply:
Yes, I only left this line in order to keep the existing functionality.

Reply:
I would rather have removed it entirely, but I suspect someone might be relying on it.

Reply:
@Ajpantuso ^

Reply:
I will clarify my statement. I think that this is the only place where type `connection.slave-type` should be set for `wifi` connections. Using the `*-slave` type parameter combined with `ssid` to mean a WiFi client connection of type `*` is a confusing interface to me as it does not have a logical connection to `nmcli`. I am requesting that an additional module option (maybe `client-type`) which maps to `connection.slave-type` be added so users only have to specify the type as `wifi` for WiFi connections. And ideally that new option would also be able to be used with `bond`, `bridge`, and `team` types as well since the `*-slave` options are being deprecated from `nmcli` at some point anyway so users may not even understand their usage in the future.